src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java

```diff
@@ -1,86 +1,53 @@
-    public String sendPunchOutRequest(String request, String punchoutUrl) {
-        LOG.debug("sendPunchOutRequest() started");
-
-        try {
-            URL url = new URL(punchoutUrl);
-            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-
-            conn.setDoInput(true);
-            conn.setDoOutput(true);
-            conn.setRequestMethod("POST");
-			String charSet = "UTF-8";
-			if (request.contains("MIME_BOUNDARY_FOR_ATTACHMENTS")) {
-				// KFSPTS-794 : for attachments
-				conn.setRequestProperty("Content-type", "multipart/related; boundary=" + CUPurapConstants.MIME_BOUNDARY_FOR_ATTACHMENTS);
-				charSet = "ISO-8859-1";
-				LOG.info("content-type is multipart/related; boundary=" + CUPurapConstants.MIME_BOUNDARY_FOR_ATTACHMENTS);
-			} else {
-				conn.setRequestProperty("Content-type", "text/xml");
-
-				LOG.info("content-type is text/xml");
-			}
-
-            OutputStream out = conn.getOutputStream();
-            OutputStreamWriter outw = new OutputStreamWriter(out, charSet);
-            outw.write(request);
-            outw.flush();
-            outw.close();
-            out.flush();
-            out.close();
-
-            InputStream inp = conn.getInputStream();
-
-            StringBuffer response = new StringBuffer();
-            int i = inp.read();
-            while (i >= 0) {
-                if (i >= 0) {
-                    response.append((char) i);
-                }
-                i = inp.read();
-            }
-            return response.toString();
-        }
-        catch (MalformedURLException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
-        catch (ProtocolException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
-        catch (UnsupportedEncodingException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
-        catch (IOException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
+    public String sendPunchOutRequest(final String request, final String punchoutUrl) {
+    	 LOG.debug("sendPunchOutRequest() started");
+
+         try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+             final HttpPost httpPost = new HttpPost(punchoutUrl);
+             if (request.contains("MIME_BOUNDARY_FOR_ATTACHMENTS")) {
+            	// KFSPTS-794 : for attachments
+                 httpPost.addHeader(HttpHeaders.ACCEPT, MediaType.MULTIPART_RELATED);
```

Comment:
Good question. The old base code was only setting Content-type but the new base code is setting both the Accept header and the Content-type which is why I updated our customization to set both. Since the unit tests pass I thought the punchout service worked fine but I may have missed something.

Reply:
We could add this header for now, but we'll need to make sure the punchout process is tested appropriately, and then make follow-up fixes as needed.

Also, I see that `MediaType` is a Spring-specific class, but the HTTP Core 5 javadocs mention that this `addHeader()` method just uses the value's `toString()` content, so I think this setup is okay as long as the string value is correct.

Reply:
I removed the accept header. I we find we need it we can add it back.